src/topology/sheaves/punit.lean

```diff
@@ -0,0 +1,38 @@
+import topology.sheaves.sheaf
+import topology.sheaves.sheaf_condition.opens_le_cover
+
+namespace category_theory
+
+open Top topological_space category_theory.limits opposite
+
+universes u v w
+
+variables {C : Type u} [category.{v} C]
+variables [Π (U : opens (Top.of (punit : Type w))), decidable (punit.star ∈ U)]
+
+lemma presheaf_on_unit_is_sheaf_of_is_terminal (F : presheaf C (Top.of (punit : Type w)))
+  (it: is_terminal $ F.obj $ op ⊥) : F.is_sheaf := λ c U s hs,
+begin
```

Comment:
I proved the forward direction @alreadydone.
I want the most general result on universe levels so I just proved everything from definition

Reply:
This is nice! I think you can write `punit.{w}`. The backward direction is [Top.sheaf.is_terminal_of_empty](https://leanprover-community.github.io/mathlib_docs/topology/sheaves/sheaf_condition/sites.html#Top.sheaf.is_terminal_of_empty) (no need of indiscrete topology), and although it has a universe constraint, the lemma it is derived from ([category_theory.Sheaf.is_terminal_of_bot_cover](https://leanprover-community.github.io/mathlib_docs/category_theory/sites/sheaf.html#category_theory.Sheaf.is_terminal_of_bot_cover)) doesn't have.

Reply:
`punit.{w}` doesn't work on me, Lean complains the following:
```
type mismatch at application
  of punit
term
  punit
has type
  Sort u : Type u
but is expected to have type
  Type ? : Type (?+1)
```